pkg/governance-scripts/contracts/20230109-gauge-adder-migration-v2-to-v3/GaugeAdderMigrationCoordinator.sol

```diff
@@ -0,0 +1,163 @@
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeAdder.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IOptimismGasLimitProvider.sol";
+
+import "../BaseCoordinator.sol";
+
+contract GaugeAdderMigrationCoordinator is BaseCoordinator {
+    IGaugeAdder public immutable newGaugeAdder;
+    IGaugeAdder public immutable oldGaugeAdder;
+
+    IGaugeController public immutable gaugeController;
+    IAuthorizerAdaptor public immutable authorizerAdaptor;
+
+    ILiquidityGaugeFactory public immutable arbitrumRootGaugeFactory;
+    ILiquidityGaugeFactory public immutable optimismRootGaugeFactory;
+
+    address public immutable liquidityMiningCommitteeMultisig;
+    address public immutable gaugeCheckpointingMultisig;
+
+    constructor(
+        IAuthorizerAdaptorEntrypoint authorizerAdaptorEntrypoint,
+        IGaugeAdder _newGaugeAdder,
+        IGaugeAdder _oldGaugeAdder,
+        ILiquidityGaugeFactory _arbitrumRootGaugeFactory,
+        ILiquidityGaugeFactory _optimismRootGaugeFactory,
+        address _liquidityMiningCommitteeMultisig,
+        address _gaugeCheckpointingMultisig
+    ) BaseCoordinator(authorizerAdaptorEntrypoint) {
+        newGaugeAdder = _newGaugeAdder;
+        oldGaugeAdder = _oldGaugeAdder;
+        arbitrumRootGaugeFactory = _arbitrumRootGaugeFactory;
+        optimismRootGaugeFactory = _optimismRootGaugeFactory;
+        liquidityMiningCommitteeMultisig = _liquidityMiningCommitteeMultisig;
+        gaugeCheckpointingMultisig = _gaugeCheckpointingMultisig;
+
+        gaugeController = _newGaugeAdder.getGaugeController();
+        authorizerAdaptor = authorizerAdaptorEntrypoint.getAuthorizerAdaptor();
+    }
+
+    // Coordinator Setup
+
+    function _registerStages() internal override {
+        _registerStage(_firstStage);
+    }
+
+    function _firstStage() private {
+        _grantPermissionsOverBridgeParameters();
+        _setupNewGaugeAdder();
+        _deprecateOldGaugeAdder();
+    }
+
+    function _afterLastStage() internal virtual override {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        authorizer.renounceRole(authorizer.DEFAULT_ADMIN_ROLE(), address(this));
+    }
+
+    // Internal functions
+
+    function _grantPermissionsOverBridgeParameters() private {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        authorizer.grantRole(
+            IAuthentication(address(arbitrumRootGaugeFactory)).getActionId(
+                IArbitrumFeeProvider.setArbitrumFees.selector
+            ),
+            gaugeCheckpointingMultisig
+        );
+        authorizer.grantRole(
+            IAuthentication(address(optimismRootGaugeFactory)).getActionId(
+                IOptimismGasLimitProvider.setOptimismGasLimit.selector
+            ),
+            gaugeCheckpointingMultisig
+        );
+    }
```

Comment:
Looks like `0x02f35dA6A02017154367Bc4d47bb6c7D06C7533B` (gauge checkpointer multisig) already has these permissions so we can remove them